src/multidim/intervalbox.jl

```diff
@@ -1,12 +1,22 @@
 # This file is part of the IntervalArithmetic.jl package; MIT licensed
 
-doc"""An `IntervalBox` is an $N$-dimensional rectangular box, given
-by a Cartesian product of $N$ `Interval`s.
-"""
 
 
-immutable IntervalBox{N,T} <: StaticVector{Interval{T}}
-    data::NTuple{N,Interval{T}}
+if VERSION >= v"0.6.0-dev"
+
+    doc"""An `IntervalBox` is an $N$-dimensional rectangular box, given
+    by a Cartesian product of $N$ `Interval`s.
+    """
+    immutable IntervalBox{N,T} <: StaticVector{N, Interval{T}}
+        data::NTuple{N,Interval{T}}
+    end
+else
+    doc"""An `IntervalBox` is an $N$-dimensional rectangular box, given
+    by a Cartesian product of $N$ `Interval`s.
+    """
+    immutable IntervalBox{N,T} <: StaticVector{Interval{T}}
+        data::NTuple{N,Interval{T}}
+    end
 end
```

Comment:
It seems odd to me that for Julia 0.5 and 0.6, the parameterization of `StaticVector` is different, though I admit that tests pass in Travis (but not locally in my machine, for Julia 0.6). 

Is this related with the requirement of StaticArrays 0.3? I think StaticArrays v0.4.0 deals with some syntax improvements related to Julia 0.6 changes. Shouldn't we change `REQUIRE` to reflect what is needed of StaticArrays?